os/board/common/partitions.c

```diff
@@ -63,6 +63,90 @@ FAR struct mtd_dev_s *mtd_initialize(FAR struct mtd_geometry_s *geo)
 #endif
 	return mtd;
 }
+
+void type_specific_initialize(FAR struct mtd_dev_s *mtd_part, int partno, const char *types)
+{
+#ifdef CONFIG_MTD_FTL
+		if (!strncmp(types, "ftl,", 4)
+#ifdef CONFIG_FS_ROMFS
+		|| !strncmp(types, "romfs,", 6)
+#endif
+#ifdef CONFIG_BINARY_MANAGER
+		|| !strncmp(types, "kernel,", 7) || !strncmp(types, "bin,", 4)
+#endif
+		) {
+			if (ftl_initialize(partno, mtd_part)) {
+				lldbg("ERROR: failed to initialise mtd ftl errno :%d\n", errno);
+				return;
+			}
+		} else
+#endif
+
+#ifdef CONFIG_MTD_CONFIG
+		if (!strncmp(types, "config,", 7)) {
+			mtdconfig_register(mtd_part);
+		} else
+#endif
+#if defined(CONFIG_MTD_SMART) && defined(CONFIG_FS_SMARTFS)
+		if (!strncmp(types, "smartfs,", 8)) {
+			char partref[4];
+
+			snprintf(partref, sizeof(partref), "p%d", partno);
+			smart_initialize(CONFIG_FLASH_MINOR, mtd_part, partref);
+		} else
+#endif
+		{
```

Comment:
Is this needed?

Reply:
I think it is not necessary.
I will modify this.